### PR TITLE
Rycar/multi os infra

### PR DESCRIPTION
**Pull request by NickRycar**
Allows for conditional installation of components based on platform of infranode, and build process will now look for a "windows" platform variable, and use a different template if it detects it, so that nix/win specific content can be separated.

### CODE REVIEW

File: generator_files/cookbooks/infranodes/recipes/default.rb [hunk 1]
```diff
@@ -54,4 +56,6 @@
 node.set['push_jobs']['chef']['node_name'] = node['demo']['node-name']
 node.default['push_jobs']['allow_unencrypted'] = true
 
-include_recipe 'push-jobs'
+if node['platform'] == 'linux'
```
Comment (andrewelizondo) [inline generator_files/cookbooks/infranodes/recipes/default.rb]: Ideally this should be:
`include_recipe 'push-jobs' if node['platform'] == 'linux'`
https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier

Comment (NickRycar) [reply generator_files/cookbooks/infranodes/recipes/default.rb]: Aaaaaandreeeeeeee!!! (fixed. :-D )

File: lib/wombat/build.rb [hunk 2]
```diff
@@ -201,8 +201,13 @@ def build_hash
         if template_name =~ /infranodes/
           infranodes.each do |name, _rl|
             next if name.empty?
+            if wombat['infranodes'][name]['platform'] == "windows"
+              infra_template = template_name + '-windows'
```
Comment (binamov) [inline lib/wombat/build.rb:205]: windows infranode support? nice! :shipit:

**Review by binamov — APPROVED** [review]
:shipit:

**Pull request merged**